File: docs/code-examples/view_coordinates_simple.cpp
// Change the view coordinates for the scene.

#include <rerun.hpp>

#include <cmath>
#include <numeric>

int main() {
    auto rec = rerun::RecordingStream("rerun_example_view_coordinates");
    rec.connect("127.0.0.1:9876").throw_on_failure();

    rec.log("world", rerun::ViewCoordinates::RIGHT_HAND_Z_UP); // Set an up-axis
    rec.log(
        "world/xyz",
        rerun::Arrows3D::from_vectors({{1.0, 0.0, 0.0}, {0.0, 1.0, 0.0}, {0.0, 0.0, 1.0}}
        ).with_colors({{255, 0, 0}, {0, 255, 0}, {0, 0, 255}})
    );
}

File: docs/code-examples/view_coordinates_simple.rs
//! Change the view coordinates for the scene.
use rerun::{
    archetypes::{Arrows3D, ViewCoordinates},
    RecordingStreamBuilder,
};

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let (rec, storage) = RecordingStreamBuilder::new("rerun_example_view_coordinates").memory()?;

    rec.log_timeless("world", &ViewCoordinates::RIGHT_HAND_Z_UP)?; // Set an up-axis
    rec.log(
        "world/xyz",
        &Arrows3D::from_vectors(
            [[1.0, 0.0, 0.0], [0.0, 1.0, 0.0], [0.0, 0.0, 1.0]], //
        )
        .with_colors([[255, 0, 0], [0, 255, 0], [0, 0, 255]]),
    )?;

    rerun::native_viewer::show(storage.take())?;
    Ok(())
}
